app/assets/javascripts/govuk_publishing_components/analytics-ga4/gtm-click-tracking.js

```diff
@@ -0,0 +1,48 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  'use strict'
+
+  function GtmClickTracking (module) {
+    this.module = module
+    this.trackingTrigger = 'data-gtm-event-name' // elements with this attribute get tracked
+  }
+
+  GtmClickTracking.prototype.init = function () {
+    var trackClicksOn = [this.module]
+    if (!this.module.getAttribute(this.trackingTrigger)) {
+      trackClicksOn = this.module.querySelectorAll('[' + this.trackingTrigger + ']')
+    }
+
+    for (var i = 0; i < trackClicksOn.length; i++) {
+      trackClicksOn[i].addEventListener('click', this.trackClick.bind(this))
+    }
+  }
+
+  GtmClickTracking.prototype.trackClick = function (event) {
+    if (window.dataLayer) {
+      var target = event.currentTarget
+      var data = {
+        event: 'analytics',
+        event_name: target.getAttribute('data-gtm-event-name'),
+        link_url: window.location.href.substring(window.location.origin.length),
```

Comment:
I _think_ this gets the path of the URL - is it worth adding a small comment to add context and explain what this does?

Reply:
Thanks, added.